lib/env.ts

```diff
@@ -1,42 +1,42 @@
 import { Option, fromNullable } from "fp-ts/lib/Option";
 import * as t from "io-ts";
 
-export const EnvKey = t.keyof({
-  SPID_SESSION_TOKEN: t.string,
-  IO_BACKEND_HOST: t.string,
-  IO_BACKEND_BASEPATH: t.string
-});
-
-export type EnvKey = t.TypeOf<typeof EnvKey>;
-
-export const EnvMap = t.record(EnvKey, t.string);
-export type EnvMap = Partial<t.TypeOf<typeof EnvMap>>;
+export const EnvMap = t.exact(
+  t.partial({
+    SPID_SESSION_TOKEN: t.string,
+    IO_BACKEND_HOST: t.string,
+    IO_BACKEND_BASEPATH: t.string
+  })
+);
+export type EnvMap = t.TypeOf<typeof EnvMap>;
 
 const pickEnvKeys = (obj: any) =>
```

Comment:
can we replace this whole module with a call to `EnvMap.decode(process.env)`  when needed ?

Reply:
You mean exposing 
```ts
export const env: EnvMap = { ...pickEnvKeys(process.env) };
```
and use it like 
```ts
const host = env.IO_BACKEND_HOST;
env.IO_BACKEND_HOST = askSessionToken();
```
?

Reply:
Is the same think we're discussing [here](https://github.com/pagopa/io-api-testing/pull/1#discussion_r387803853)?

Reply:
i mean `export const config = EnvMap.decode(process.env).getOrElse(...)`

it's not clear why do you need to reassign

```
const host = env.IO_BACKEND_HOST;
env.IO_BACKEND_HOST = askSessionToken();
```

can't you jst use `const sessionToken = askSessionToken()` where's needed ?

Reply:
no, because I need such information in another module of another workflow (the jest test runner)

Reply:
Now I get it ! You are into an edge case where you need a global setup for jest that takes input from command line (if I remember well one of the initial suggestion was just take the values from environment and drop the cli prompts as tests usually runs inside CI environment). You probably need some jest facility like this: https://jestjs.io/docs/en/configuration.html#globalsetup-string
but it depends on [jest support](https://github.com/facebook/jest/issues/7184). I renew my suggestion to just use environment variables for configuration.

Reply:
`globalSetup` is what I'm using. The difference from the given example is that I built a custom global store instead of accessing the `global` object directly. I did it for implementing the type-driven logic we long discussed (`type EnvMap`).

**About the prompt:**  
The prompt itself is just a toy, I think we'll use it very rarely. What I do care about is the architecture is built upon, which may enable near-future improvements to the test suite like data seeding, data enrichment, automatic token retrieval, data cleanup, service start and teardown, etc.
If the cost is to have an assignment in a module, I advocate we keep it as it might make such feature easier to implement. Yes, this is very speculative and so close to _premature optimization_ pitfall. I leave the choice to you thou. 

If you are worrying about having the service startup complicates, please consider that the jest runner itself is build to only care about env variables. So if we want to get rid of the setup part we can just remove `globalSetup: './setup/index.ts'` from `jest.config.js`, and the test suite keeps working as expected.

Reply:
about 

> I think we'll use it very rarely

the suggestion is [this](https://it.wikipedia.org/wiki/You_aren%27t_gonna_need_it). 
best code is no code. please remove the code about prompts.

Reply:
Ok.